pages/[posts]/[slug]/[page_number].tsx

```diff
@@ -0,0 +1,120 @@
+import fs from 'node:fs';
+import type { GetStaticPaths, GetStaticProps } from 'next';
+import type { DualPost, Post } from '../../../lib/types';
+import PostPreview from '../../../components/PostPreview';
+import { getAllPosts } from '../../../lib/getAllPosts';
+import { podcastXml } from '../../../lib/podcast';
+import PageWrapper from '../../../components/PageWrapper';
+import Paginator from '../../../components/Paginator';
+import { paginate } from '../../../lib/helpers';
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const dualPosts = await getAllPosts();
+  const whichLanguage = context.params?.slug === 'page' ? 'en' : 'es';
+  const posts = dualPosts.map((post) => ({
+    ...post[whichLanguage],
+    category: post.category,
+  }));
+  const whichPage = Number(context.params?.page_number);
+  const thisPagePosts = paginate(posts, whichPage, 8);
+
+  return {
+    props: {
+      posts: thisPagePosts,
+      pageCount: Math.ceil(posts.length / 8),
+      pageNum: whichPage,
+    },
+  };
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const allDualPosts = await getAllPosts();
+
+  const enPosts = allDualPosts.map((dual) => dual.en);
+  const esPosts = allDualPosts.map((dual) => dual.es);
+  fs.writeFileSync(`./public/podcast.en.rss`, podcastXml(enPosts));
+  fs.writeFileSync(`./public/podcast.es.rss`, podcastXml(esPosts));
+
+  const numberOfPages = Math.ceil(allDualPosts.length / 8);
+  const pages = new Array(numberOfPages).fill(0).map((_, index) => index + 1);
+
+  const paths = pages
+    .map((page) => ({
+      params: { posts: 'posts', slug: 'page', page_number: String(page) },
+    }))
+    .concat(
+      pages.map((page) => ({
+        params: {
+          posts: 'publicaciones',
+          slug: 'pagina',
+          page_number: String(page),
+        },
+      })),
+    );
+
+  return { paths, fallback: false };
+};
+
+interface Props {
+  posts: Array<Post<'en' | 'es'> & { category: 'teaching' | 'post' }>;
```

Comment:
maybe pull in `Lang` here?

...and i thought `category` was on the `Post` object already.  if it's not, why isn't it?